# HW 6

Just generic python setup first, including some helper functions to help with degree-radians conversion, and solving quadratic equations

In [1]:
import numpy as np
from numpy import sin, cos, tan, arctan2, sqrt

In [2]:
def rad(th):
    return th * np.pi/180

def deg(th):
    return th * 180/np.pi

def quadratic(a, b, c):
    return ((-b + sqrt(b**2 - 4*a*c)) / (2*a),
            (-b - sqrt(b**2 - 4*a*c)) / (2*a))

## Problem 2

Let's get the fun started with a sketch and 3 FBDs:

![figure](hw6_2.png)

I chose the coordinate system for each block such that the $x$ direction goes in the direction that the block actually moves, so acceleration will be positive, and the $y$ direction perpendicular to that.

The $60^\circ$ angle that is given is kinda unusual, but obviously the remaining angle of the slope of the horizontal is easily found to be $\theta = 30^\circ$, so I'll go with that.

So we have to set up Newton's 2nd Law for each block. For the first block, there's only one non-trivial direction:
$$
m_1 a_1 = F_{net, on 1} = m_1 g - T_1
$$

For the other two blocks, we'll need to look at the $x$ and $y$ directions:
\begin{align}
m_2 a_2 &= F_{net, \,on\, 2, x} = 2 T_2 + m_2 g \sin\theta - F_{f, 3\, on\, 2}\\
0 &= F_{net, \,on\, 2, x} = F_{N, 3\,on\,2} - m_2 g \cos\theta\\
m_3 a_3 &= F_{net, \,on\, 3, x} = T_1 - T_2 - m_3 g \sin\theta - F_{f, 2\, on\, 3} - F_{f, T\,on\,3}\\
0 &= F_{net, \,on\, 3, x} = F_{N, T\,on\,3} - F_{N, 2\,on\,3} - m_3 g \cos\theta\\
\end{align}

Let's think about accelerations for a minute: As the top block (2) slides $1 cm$, the bottom block (3) slides $2 cm$, as will the weight (1). Therefore
$$a_1 = a_3 = 2a_2.$$

I will simply use $a$, where $a_1 = a_3 = a$ and $a_2 = \frac{1}{2} a$.

It's all algebra from here. From the $y$ component of block 2:
$$F_{N, 3\,on\,2} = m_2 g \sin\theta$$

$F_{N, 2\,on\,3}$ is the 3rd law pair partner, hence has the same magnitude:
$$F_{N, 2\,on\,3} = m_2 g \cos\theta$$

From the $y$ component of block 3, and using the normal force we just found:
$$F_{N, T\,on\,3} = (m_2 + m_3) g \cos\theta$$

Now we still gotta work through the direction with actual acceleration. I'm writing them again using $a$, and the equation for block 2 I'll multiply by \frac{1}{2}, so that the $2T_2$ becomes a single $T_2$ because I want that to cancel out next:
\begin{align}
m_1 a &= m_1 g - T_1\\
\frac{1}{4} m_2 a &= T_2 + \frac{1}{2}m_2 g \sin\theta - \frac{1}{2}F_{f, 3\, on\, 2}\\
m_3 a &= T_1 - T_2 - m_3 g \sin\theta - F_{f, 2\, on\, 3} - F_{f, T\,on\,3}\\
\end{align}
Now I can just add all three equations to make $T_1$ and $T_2$ cancel out. I'll also replace the friction forces by the respective $\mu F_N$.

$$
(m_1 + \frac{1}{4}m_2 + m_3) a =
(m_1 - (m_3 - \frac{1}{2} m_2) \sin\theta)g - \mu(\frac{3}{2}m_2 + m_2 + m_3)g\cos\theta
$$

I can solve for $a$:

$$
 a =
\frac{(m_1 - (m_3 - \frac{1}{2} m_2) \sin\theta) - \mu(\frac{3}{2}m_2 + m_2 + m_3)\cos\theta}
{m_1 + \frac{1}{4}m_2 + m_3}g
$$

and plug in numbers.

In [3]:
m1 = 10
m2 = 2
m3 = 3
mu = 0.3
d = 0.5
theta = rad(90. - 60.)
g = 9.8

a = (m1 - (m3 - .5*m2)*sin(theta) - mu * (1.5*m2 + m2 + m3) * cos(theta)) / (m1 + m3 + .25*m2) * g
print("a = {:f}".format(a))

a = 5.024525


As the top block (2) moves to the right/down by $(1/3) \cdot 50$ cm, the bottom block (3) moves left/up by $(2/3) \cdot 50$ cm, so that's where they meet.

So
$$\frac{2}{3} d = \frac{1}{2} a t^2 \Longrightarrow t = \sqrt{\frac{4d}{3a}}.$$

In [4]:
t = sqrt((4*d)/(3*a))
print("t = {:f}".format(t))

t = 0.364256


Final answer: It takes 0.36 seconds for the edges to line up.